src/Fantomas/CodePrinter.fs

```diff
@@ -2023,14 +2023,14 @@ and genExpr astContext synExpr ctx =
 
                 atCurrentColumn (colWithNlnWhenItemIsMultilineUsingConfig items) ctx
         // Could customize a bit if e is single line
-        | TryWith (e, cs) ->
+        | TryWith (e, mWithToLast, cs) ->
             atCurrentColumn (
                 kw TRY !- "try "
                 +> indent
                 +> sepNln
                 +> genExpr astContext e
                 +> unindent
-                +> kw WITH !+~ "with"
+                +> tokN mWithToLast WITH !+~ "with"
```

Comment:
Close, but this still doesn't cover all cases:
```fsharp
[<Test>]
let ``nested try/with with comment on with`` () =
    formatSourceString
        false
        """
try
    a
with
| b ->
    try c
    // inner comment
    with
    | d -> ()
"""
        config
    |> prepend newline
    |> should
        equal
        """
try
    a
with
| b ->
    try
        c
    // inner comment
    with
    | d -> ()
"""
```

This will still fail, unfortunately. ([online tool](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAXATgTwDqdwDsIIBDfIgdwEt0ALCwgHzACMwBaAPkZLC2xhovEgHpRYaoUKJMQ2AFsFiQuhEQa9dWBbxOXMAAoAlJAA04KAGdE6dFIDmV1GADaliKD4RIAEnTYAA7IaJBS6OYeJH4AbqQANgCuIW5RfAAMFt7RAJKESKoAytQAXshZ2WAALGlgALpRAL4VnrV%2BAcEuYaqR2bEJyS7ulWAAjC3ekACypAAeADJSiPMqDvS9laMATOm1DXzNUV6T-kEpkKywsPEbfP1JKcOVWxN3hYgK1HDxsIQAgugAKL5ADyADNFjJbt4wQkbHsmhNjndTp1QpdrtCfL44g8hrUIABmV7RQqBUjQRAAIUQYNgmEQAAVSJhSMp0LIsSQsMkEQckW1UecMTcSfdBmgntkqiSfGSKdTafTlrBKLJoKQbHkYrANfZflyIDzEHySIc%2BMjokKuiKueLHgSwABWWVQeWUml0hkAVUCwUwGq1hB1euoBtdsPi8Oy%2BzNAr61vRV1FgtxEtSIwAbK7IO7FV7EABheKaqyF35WHnQdD0w1gSPR7yxiDmkiW7EdYXJu04gYOkYAdhzec9yqmH1YnIjcJNMcRR0FnZt3bFvbxksdAA5h%2BSPUqGeWfoQ6w3Z035xbF2dl5jV2n%2B5UAJw7hV-MEczDlpSkOvG00t%2BMTiXJNb1TPt8RGUZMkdXNd3zZV3k%2Bb5w0dU9-zAVtWgTYCoFtO9wI3SDxhgvM3w-RCvmuFCRj-Od%2BQXbDrxAlME3vCDNheGC8gKdAQUIAAVHAAHVaDoE8Z3QzCwHbKBE1wlcwPXDNNmJEi4L%2BTBYESfIABFEHiahPg-X9MF5Oi4wYoCmKgcIezYwjNhlGCZlmHIwX4ugVEBKNEEKOh6XQET4HWV0ql2cyAMslEcO6CJ8KUqVvFGF1nLmPIwWoWYQX9Uga0wQFZkCBkrCsMNj1dJ1wvPejL0YtEbJ6eL00SvhRmzVLZgAJUQOBMHgIKQsdMLJMA6LrNiuyCOU7JRiHDrut6%2BAADlEgUSdMHBHIOQUZxXVGEaoqtGKFESeJ7AMmQADF6QUXKP3468mofGbtxghb6XgKZTvOpZrswW67CnGDoDoFkKQ-AB9GhgrEg7aqs%2BqJqe9iZufDqNNZbAQUwRZKwGsTQqqvhmww0ajvG2zkYc7IdhzFyMdILGceoSsVrW2RNu23bHX2iLScOjtxpOs7qAuxA-oB%2B7HsU5rHS2YiRkgBmmdx9AvpFsWJbuoHFZBsHq1kKHqBhyA4bbK9EcpmXnu8LZOMVlyADU%2BypKR4EcfG603ImSBJqSZPaCnGutlHbdUh25ku7TqzK138g943BpGYa%2Bf9i3zit1ippakgticiPZh02B0AAcVsAqisQEqys9iqfYgP2ycFxG8JD6nbZSh3vtFpYqR%2BaAAGsqVZQfbCsPjCjZItrlW8rUIk1Om9kmLW6zhK5faxWlsQSgxZpdA1RUB7gj0jLCFoMq-nycd2cwbmRjQxeBeX8bV5OezpttubFYAaUQRBAhuQ8ioPIk9lCQnKPPHyZssIIwzsHNessRhbFegXbynwrB0FrkNeu9QLzmzql2UCiCbZ8C2GjRWxRCAOHiIgDSDhVoqECogVgUwECQIfgvaqFl4ZjRbgpEhoc%2BCEmgorP4BkHCECjoQGOvxiiSNyokBk-FYDcSYblMq4loFP14eTfhxD37Z0dISBWlQlZnVkIQDRMQVTUJvutU%2BUgL4Vi0Y2Ym%2BDYF8KISxQx68RiEntmY9W9gIHzDZKweApBiywFKtQ7eu8liuLPO4mqBC4E3h8SiD%2BOciThzMTpFmpBWC0PQSzOghRsCqDmEkmB0l05dBUPAcEEDCgBFoZNPxlRCT5zMcCJpEJEk5mgJgeIYJTY6LSV4jJHSkFdM7mYv%2BAC1GqDyMPUgMiCZQLcb7DxdTCHTKpp-YRm8zF93WQPCBVgNJaXyNvSsiBPrdzFhXYqpUXGulotwyKujm7eJmaQkghJv6nJZKOBkBSrDDMMlIXKDzvTn1%2BHpaAJZWT6jnpw7RXz%2BY-JfvozJVpsnGNQWY1pmAvhq3YTU1Olg6gVDCFYS6pUXBoUaEAA))

Reply:
I updated the code and test cases. Please take a look when you find some free time 🙏